R/target_market_share_company.R

```diff
@@ -97,9 +97,27 @@ target_market_share_company <- function(data) {
       )
     ) %>%
     tidyr::pivot_longer(
-      cols = tidyr::ends_with("weighted_production"),
-      names_to = "production_name",
-      values_to = "production_value"
+      cols = c("tmsr_target_weighted_production", "smsp_target_weighted_production"),
```

Comment:
Just noticing that these two files seem to have common patterns that we might be able to DRY:

* R/target_market_share_company.R
* R/target_market_share_portfolio.R

Reply:
Ya absolutely, let's open this as a new issue and discuss, prior to submitting to CRAN. Overall I think there are a couple of places we could do some housekeeping in this package. 
In particular, `summarize_portfolio_production` and `summarize_company_production` both exist, because I think both outputs are useful to the banks. But because we have two pathways, we have two target functions etc. etc. 

Likely we could just add a `...` argument to the function, and allow for a "company" grouping variable, to achieve both results, and then add a vignette to explain how to do this.